### PR TITLE
fix go vet issues, add go vet to CI

### LINKED ISSUE #349: go vet issues
remote.go:44: wrong number of args for format in Sprintf call: 2 needed but 3 args
plumbing/format/packfile/scanner.go:304: method Seek(offset int64) (previous int64, err error) should have signature Seek(int64, int) (int64, error)
plumbing/format/pktline/scanner_test.go:221: Println call ends with newline
plumbing/object/change_adaptor.go:18: no formatting directive in Errorf call
plumbing/object/change_adaptor.go:22: no formatting directive in Errorf call
plumbing/object/change_adaptor.go:55: missing argument for Errorf("%s"): format reads arg 2, have only 1 args
plumbing/object/change_test.go:241: unreachable code
utils/merkletrie/difftree_test.go:123: arg c.Action in printf call is a function value, not a function call
utils/merkletrie/iter_test.go:84: arg t.operations[i] for printf verb %s of wrong type: byte
utils/merkletrie/internal/fsnoder/new.go:37: missing argument for Errorf("%c"): format reads arg 1, have only 0 args

### PR DESCRIPTION
**Pull request by smola**
Fixes https://github.com/src-d/go-git/issues/349

### CODE REVIEW

File: plumbing/format/packfile/scanner_test.go [hunk 1]
```diff
@@ -4,8 +4,8 @@ import (
 	"bytes"
 	"io"
 
-	. "gopkg.in/check.v1"
 	"github.com/src-d/go-git-fixtures"
+	. "gopkg.in/check.v1"
```
Comment (mcuadros) [inline plumbing/format/packfile/scanner_test.go]: mix of internal and external imports

File: plumbing/format/packfile/scanner.go [hunk 2]
```diff
@@ -301,7 +301,7 @@ func (s *Scanner) copyObject(w io.Writer) (int64, error) {
 }
 
 // Seek sets a new offset from start, returns the old position before the change
```
Comment (ajnavarro) [inline plumbing/format/packfile/scanner.go]: You miss to change documentation

Comment (smola) [inline plumbing/format/packfile/scanner.go]: :+1:

File: plumbing/format/packfile/scanner_test.go (hunk 1 above)
Comment (smola) [inline plumbing/format/packfile/scanner_test.go]: Fixed.

**Pull request merged**